sdk/keyvault/azure-security-keyvault-administration/CMakeLists.txt

```diff
@@ -39,9 +39,9 @@ if(FETCH_SOURCE_DEPS)
         add_subdirectory(${oneFolder})
     endforeach()
 elseif(NOT AZ_ALL_LIBRARIES)
-  find_package(azure-core-cpp "1.7.2" CONFIG QUIET)
```

Comment:
What's the harm in specifying these version strings for `find_package()` within the CMakeLists.txt and in `find_dependency()`?

Is it just that we'd have 2 places to keep up-to-date, or is there some correctness/functionality bug?

Essentially, what's the value of this cleanup, to the end customer?

Reply:
It does not support semvers, i.e. things like "1.0.0-beta.1", so this broke us several times when people were updating it, and it is completely unnecessary. Also this always needs to lag behind the released version, because if we update it in the repo with the version that is not yet merged to vcpkg, the build is going to fail, we can only increment package version here after it ia available in the public vcpkg.
I.e. this is costly, error-prone, does not do well everything that we need from it, and so far it has only been the source of problems. If we want an additional mechanism, we need the correct one.